### ISSUE TITLE
[a11y]: ComposedModal dialog body content is not read out by JAWS

### ISSUE THREAD

**Issue by andreas-schmitt-mi** [carbon-design-system/carbon#10089]
### Package

carbon-components, carbon-components-react

### Browser

Firefox

### Operating System

Windows

### Package version

v10.47.0, v7.47.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Assistive technology

JAWS 2021.2103.174

### Description

JAWS screen reader tool only reads out modal dialog label and primary button text when the dialog is opened. The content of the modal body is not read out. Maybe I'm missing something, but I could not find a way to navigate JAWS to the dialog content e.g. using tab key. This might make it difficult for people with screen readers to understand what the dialog is about.

I noticed something strange however, if I modify the CodeSandbox so that the dialog is displayed immediately e.g. `<ComposedModal open={true}>` then the whole dialog including the modal body is read out by JAWS.

Not sure if this could be related to https://github.com/carbon-design-system/carbon/issues/2739

### CodeSandbox example

https://codesandbox.io/s/loving-volhard-23c89?file=/src/index.js:364-377

### Steps to reproduce

1. Open CodeSandbox.
2. Open JAWS and open "Launch dialog" button.
3. JAWS reads `<ModalHeader>` label "Task failed" and  the "Yes" button. The title of the `<ModalHeader>` and the content of the `<ModalBody>` is not read out by JAWS.
4. Modify CodeSandbox `<ComposedModal open={true}>`
5. JAWS reads the whole dialog including the modal body.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by gcattan** [comment]
Hello, we experienced the same issue with Modal. Is there any update concerning this issue?

**Comment by gcattan** [comment]
@joshblack @abbeyhrt I am not sure about the status of this issue. Could you please help me?

**Comment by tay1orjones** [comment]
@gcattan Sorry for the lack of response, I've added this to our next sprint for someone to take a closer look.

**Comment by dakahn** [comment]
I'm unable to reproduce this bug in Firefox v103.0.2 on Windows 11 using JAWS v2022.2207.25ILM testing against our [v10 Storybook](https://v10-react.carbondesignsystem.com/?path=/story/components-composedmodal--with-state-manager). JAWS reads all modal headings, content, and form elements as expected. 

Can you test against our Storybook example and confirm this @andreas-schmitt-mi ?

**Comment by dakahn** [comment]
closing as a false report/resolved. if that's not the case comment and we'll reopen 👍🏽

**Comment by andreas-schmitt-mi** [comment]
@dakahn Sorry for the late response as I was on vacation. Yes, I can still reproduce that issue with the v10 Storybook example with Firefox 104 on Windows 11 and JAWS 2021.2111.13 ILM. Only parts of the dialog are read out by JAWS. Dialog title "Add a custom domain" and dialog text are not read out. Unfortunately I can not try it with a more recent version of JAWS.

**Comment by andreas-schmitt-mi** [comment]
@dakahn Is only the latest version of JAWS supported by carbon components?

**Comment by andreas-schmitt-mi** [comment]
@dakahn As a side note, it seems that the issue described for JAWS 2021 also occurs on VoiceOver of the latest MacOS 12.5.1
